### ISSUE TITLE
HTTP File Upload (UPLOAD) is there any plan to implement it ?

### ISSUE THREAD

**Issue by Omid-Mohajerani** [UWPX/UWPX-Client#149]
Hi,

Sorry if it is not the right place to ask. I wonder if there is any plan to code HTTP File Upload (UPLOAD) part? as I can see from Feb 2020 there is no update in this part? By any chance any one working on it?

**Comment by COM8** [comment]
Yes!
But I do not know when.
I hope this year.

I'm currently working on refactoring OMEMO and the DB implementation in #146 so the backend and especially OMEMO work more reliable.

After that I would like to finish Push support and then perhaps HTTP upload :)

**Comment by COM8** [comment]
As soon as #146 has been merged, there will be a new release.

**Comment by Omid-Mohajerani** [comment]
Great. thanks for the update.

**Comment by binarydad** [comment]
Any progress on this? The latest version doesn't seem to allow uploading. Thanks!

**Comment by COM8** [comment]
Yes, with the [last release](https://github.com/UWPX/UWPX-Client/releases/tag/v.0.39.0.0) I introduced user avatars. This allows me to build on top of it to offer HTTP file upload.
But currently I don't have an exact time plan for it since I'm currently working on improving push support.
